Compression/ZipStreamWriter.cs

```diff
@@ -151,7 +151,7 @@ private void CopyTempFile(int attempts, bool throwOnFailure)
 
             if (throwOnFailure)
             {
-                throw new Exception(string.Format("Unable to save file: {0} after {1} attempts.", _filename, attempts));
+                throw new Exception($"Unable to save file: {_filename} after {attempts} attempts.");
```

Comment:
Wondering whats the criteria to call `.ToStringInvariant()` on some `int` like https://github.com/QuantConnect/Lean/pull/3442/files#diff-7b732fdae7d43c1eccb84cabe6af4e21R143 `order.Id.ToStringInvariant();` but here on `attempts` not?

Reply:
The initial criteria I used was anything that raised the CA1304/CA1305 errors. _Sadly_, .net 4.5.2 doesn't raise these warnings when inside an interpolated string, which is what led to so many of these being missed. My understanding is that this is because `FormattableString` isn't natively available in .net 4.5.2, but is available in 4.6+ and dotnet core, so whenever we upgrade we should get warnings for them at that time.

_In general_, any type that implements `IConvertible` or `IFormattable` should be using `CultureInfo.InvariantCulture`, so this includes all `int`s. In order to get _all_ cases, I've begun the tedious process of going through each file and inspecting each string interpolation. If we're interpolating non-string values into the string, I'm adding `using static QuantConnect.StringExtensions` and using the `Invariant` method which is implemented via `FormattableString.Invariant` which applies the invariant culture to each argument in the interpolated string. This process has been completed for `Common`.